backend/services/implementations/classService.ts

```diff
@@ -0,0 +1,260 @@
+        const testSessionPromises = testSessionIds.map((id) =>
+          this.testSessionService.getTestSessionById(id),
+        );
+        const testSessionDTOs = await Promise.all(testSessionPromises);
+
+        return {
+          id: classObj.id,
+          className: classObj.className,
+          schoolYear: classObj.schoolYear,
+          gradeLevel: classObj.gradeLevel,
+          teacher: teacherDTO,
+          testSessions: testSessionDTOs,
+          students: classObj.students,
+        };
+      }),
+    );
+    return classDtos;
+  }
+
+  async updateClass(
+    id: string,
+    classObj: ClassRequestDTO,
+  ): Promise<ClassResponseDTO> {
+    let updatedClass: Class | null;
+    try {
+      updatedClass = await MgClass.findByIdAndUpdate(
+        id,
+        {
+          className: classObj.className,
+          schoolYear: classObj.schoolYear,
+          gradeLevel: classObj.gradeLevel,
+          teacher: classObj.teacher,
+          testSessions: classObj.testSessions,
+        },
+        {
+          new: true,
+          runValidators: true,
+        },
+      );
+
+      if (!updatedClass) {
+        throw new Error(`Class id ${id} not found`);
+      }
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to update class. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+    return (await this.mapClassToClassDTOs([updatedClass]))[0];
+  }
+
+  async deleteClass(id: string): Promise<string> {
+    try {
+      const deletedClass: Class | null = await MgClass.findByIdAndDelete(id);
+      if (!deletedClass) {
+        throw new Error(`Class with id ${id} not found`);
+      }
+      return id;
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to delete class. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async createStudent(
+    student: StudentRequestDTO,
+    classId: string,
+  ): Promise<ClassResponseDTO> {
+    let newStudent: StudentResponseDTO;
+    let classObj: Class | null;
+
+    try {
+      // create a new StudentResponseDTO
+      newStudent = {
+        id: "",
+        firstName: student.firstName,
+        lastName: student.lastName,
+        studentNumber: student.studentNumber,
+      };
+
+      classObj = await MgClass.findById(classId);
+
+      if (!classObj) {
+        throw new Error(`Class id ${classId} not found`);
+      }
+
+      classObj?.students.push(newStudent);
+      await classObj?.save();
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to create student. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+    return (await this.mapClassToClassDTOs([classObj]))[0];
+  }
+
+  async updateStudent(
+    studentId: string,
+    classId: string,
+    student: StudentRequestDTO,
+  ): Promise<ClassResponseDTO> {
+    let studentToUpdate;
+    let updatedClass: Class | null;
+    try {
+      updatedClass = await MgClass.findById(classId);
```

Comment:
```suggestion
      class = await MgClass.findById(classId);
```
rename since class is not actually updated